cognite/client/data_classes/simulators/runs.py

```diff
@@ -201,6 +206,7 @@ def __init__(
         self.data_set_id = data_set_id
         self.user_id = user_id
         self.log_id = log_id
+        self._cognite_client = cast("CogniteClient", cognite_client)
```

Comment:
Is it guaranteed that we will get a client and not None?

Reply:
this is a common pattern throughout the SDK, I assume it's not given it is available exactly on this line, but it will be injected by some internal magic.
@doctrino do you know why this is correct way of doing it, or we should remove the cast?

Reply:
Yes, it is. All `CogniteResource` (base class) uses a `_WithClientMixin`, https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/data_classes/_base.py#L90, that raises an appropriate error if the client is not there.